test/Interop/Cxx/function/Inputs/default-arguments.h

```diff
@@ -0,0 +1,55 @@
+inline void unused(int x = 0) {}
+
+inline bool isZero(int value = 0) { return value == 0; }
+inline bool isNonZero(int value = 0) { return value != 0; }
+
+inline bool isNil(int *ptr = nullptr) { return ptr == nullptr; }
+inline bool isStrNil(const char *ptr = "") { return ptr == nullptr; }
+
+inline int sum(int a, int b = 1) { return a + b; }
+inline int subtract(int a = 123, int b = 1) { return a - b; }
+
+static int counter = 0;
+inline bool isZeroCounter(int value = counter++) { return value == 0; }
+
+struct ArgTy {
+  int value;
+
+  static ArgTy createNonZero() { return {1}; }
+};
+inline bool isArgZero(ArgTy a = {0}) { return a.value == 0; }
+inline bool isArgNonZero(ArgTy a = ArgTy::createNonZero()) { return a.value != 0; }
+inline bool isArgZeroConstRef(const ArgTy &a = {0}) { return a.value == 0; }
+
+struct HasMethodWithDefaultArg {
+  bool isZero(int v = 0) const { return v == 0; }
+};
+
+struct DerivedFromHasMethodWithDefaultArg : public HasMethodWithDefaultArg {};
```

Comment:
Can you add a test for another level of derived (DerivedDerived) as there's another synthesis interaction happening past the first derived type.

Reply:
Done